modules/actions/cow-amm/update-surplus-aprs.ts

```diff
@@ -1,31 +1,71 @@
-import { PrismaPool, PrismaPoolAprType, PrismaPoolDynamicData } from '@prisma/client';
+import { PrismaPoolAprType, PrismaPoolSnapshot } from '@prisma/client';
 import { prisma } from '../../../prisma/prisma-client';
 
+const getSnapshotsByTimestamp = async (timestamp: number = Math.floor(Date.now() / 1000 / 86400) * 86400) => {
+    const snapshots = await prisma.$queryRaw<PrismaPoolSnapshot[]>`
+        SELECT DISTINCT ON ("poolId") *
+        FROM "PrismaPoolSnapshot"
+        JOIN "PrismaPool" ON "PrismaPoolSnapshot"."poolId" = "PrismaPool"."id"
+        WHERE "PrismaPool"."type" = 'COW_AMM'
+        AND "PrismaPoolSnapshot"."timestamp" = ${timestamp}
+        ORDER BY "poolId", timestamp DESC;
+    `;
+
+    return snapshots;
+};
+
 export const updateSurplusAPRs = async () => {
-    const pools = await prisma.prismaPool.findMany({
-        where: {
-            type: 'COW_AMM',
-        },
-        include: {
-            dynamicData: true,
-        },
-    });
-
-    // Filter out pool with no dynamic data
-    const surpluses = pools.filter(
-        (pool): pool is PrismaPool & { dynamicData: PrismaPoolDynamicData } => pool.dynamicData?.surplus24h !== null,
-    );
-
-    const data = surpluses.map((pool) => ({
-        id: `${pool.id}-surplus`,
+    // Find the snapshot
+    const latestSnapshots = await getSnapshotsByTimestamp();
```

Comment:
Would rather also call this with a variable instead of relying on a default value in the function

Reply:
done